### PR TITLE
devicestate: fix panic in firstboot code when no snaps are seeded

### PR DESCRIPTION
**Pull request by mvo5**
Samuele noticed that the code in firstboot will panic if there is no snap to seed. This PR fixes this.

### CODE REVIEW

File: overlord/devicestate/firstboot_test.go [hunk 1]
```diff
@@ -1065,7 +1091,7 @@ snaps:
 }
 
 func (s *FirstBootTestSuite) TestImportAssertionsFromSeedClassicModelMismatch(c *C) {
-	release.OnClassic = true
+	release.MockOnClassic(true)
```
Comment (zyga) [inline overlord/devicestate/firstboot_test.go]: Do you need to restore this?

File: overlord/devicestate/firstboot_test.go [hunk 2]
```diff
@@ -189,8 +189,34 @@ func (s *FirstBootTestSuite) TestPopulateFromSeedOnClassicNoSeedYaml(c *C) {
...
+	st.Lock()
+	defer st.Unlock()
+
+	_, err = devicestate.PopulateStateFromSeedImpl(st)
+	c.Assert(err, ErrorMatches, "cannot proceed, no snaps to seed")
+}
+
 func (s *FirstBootTestSuite) TestPopulateFromSeedOnClassicNoSeedYamlWithCloudInstanceData(c *C) {
-	release.OnClassic = true
+	release.MockOnClassic(true)
```
Comment (zyga) [inline overlord/devicestate/firstboot_test.go]: Do you need to restore this?

File: overlord/devicestate/firstboot_test.go [hunk 3]
```diff
@@ -189,8 +189,34 @@ func (s *FirstBootTestSuite) TestPopulateFromSeedOnClassicNoSeedYaml(c *C) {
 	c.Check(ds.Model, Equals, "my-model-classic")
 }
 
+func (s *FirstBootTestSuite) TestPopulateFromSeedOnClassicEmptySeedYaml(c *C) {
+	release.MockOnClassic(true)
```
Comment (zyga) [inline overlord/devicestate/firstboot_test.go]: Do you need to restore this?

File: overlord/devicestate/firstboot_test.go [hunk 4]
```diff
@@ -159,7 +159,7 @@ func (s *FirstBootTestSuite) TestPopulateFromSeedOnClassicNoop(c *C) {
 }
 
 func (s *FirstBootTestSuite) TestPopulateFromSeedOnClassicNoSeedYaml(c *C) {
-	release.OnClassic = true
+	release.MockOnClassic(true)
```
Comment (zyga) [inline overlord/devicestate/firstboot_test.go]: Also here

File: overlord/devicestate/firstboot_test.go (hunk 1 above)
Comment (mvo5) [reply overlord/devicestate/firstboot_test.go]: The firstboot suite does "	s.restoreOnClassic = release.MockOnClassic(false)" in SetUpTest and does the restore in "TearDownTest" - if you prefer this to have a restore just for easier reading I'm happy to add it.

Comment (pedronis) [reply overlord/devicestate/firstboot_test.go]: that's why it was written the other way btw

**Comment by pedronis** [comment]
given that this still has the bug about last  vs -1 it should land after #5392

File: overlord/devicestate/firstboot.go [hunk 5]
```diff
@@ -189,7 +189,9 @@ func populateStateFromSeedImpl(st *state.State) ([]*state.TaskSet, error) {
 
 	// chain together configuring core, kernel, and gadget after
 	// installing them so that defaults are availabble from gadget
-	configTss[0].WaitAll(tsAll[last])
+	if len(tsAll) > last {
```
Comment (pedronis) [inline overlord/devicestate/firstboot.go]: I would have used len(configTss) > 0  around the full 3 lines, but no strong opinion

Comment (mvo5) [reply overlord/devicestate/firstboot.go]: Indeed, I think this is a bit easier to read.

File: overlord/devicestate/firstboot_test.go (hunk 1 above)
Comment (mvo5) [reply overlord/devicestate/firstboot_test.go]: I know, I can revert its just that when I searched for classic tests I did a search for MockOnClassic() and was confused. I can revert this if you prefer the old style.

Comment (pedronis) [reply overlord/devicestate/firstboot_test.go]: I'm fine either way myself, I have sympathy for people expecting the restore in this style tough

**Pull request merged**